docs: add the example of pvc-based ceph cluster on top of bare metal (#7969)
rook/rook: Documentation/ceph-cluster-crd.md

## PVC-based Cluster

In a "PVC-based cluster", the Ceph persistent data is stored on volumes requested from a storage class of your choice.

> This type of cluster is recommended to use with CSI drivers which support dynamic volume provisioning.
> in clusters where a local PV provisioner is available.

Reviewer: ```suggestion
This type of cluster is recommended to use with CSI drivers which support dynamic volume provisioning and also
in clusters where a local PV provisioner is available.
```

> **NOTE**: Kubernetes version 1.13.0 or greater is required to provision OSDs on PVCs.

> ```yaml

Reviewer 2: I'd like to keep the examples at the top of this page simple, and have more complex examples later in the doc. What about this?
- This example should match the cluster-on-pvc.yaml example for aws
- We should add a new section later in the doc for this example that creates local PVs and uses a local provisioner

kind: StorageClass

> apiVersion: storage.k8s.io/v1

Reviewer 2: Please add this example yaml in the examples folder, for example named cluster-on-local-pvs.yaml. It's easier to use the examples if we have them in the example folder.

* `devices`: A list of individual device names belonging to this node to include in the storage cluster.
  * `name`: The name of the device (e.g., `sda`), or full udev path (e.g. `/dev/disk/by-id/ata-ST4000DM004-XXXX` - this will not change after reboots).
  * `config`: Device-specific config settings. See the [config settings](#osd-configuration-settings) below

Host-based cluster only supports raw device and partition.

> And below is the settings for PVC-based cluster.

Reviewer 2: ```suggestion
Below are the settings for a PVC-based cluster.
```
